src/bench/wallet_migration.cpp

```diff
@@ -32,41 +33,46 @@ static void WalletMigration(benchmark::Bench& bench)
     int NUM_WATCH_ONLY_ADDR = 20;
 
     // Setup legacy wallet
-    DatabaseOptions options;
-    options.use_unsafe_sync = true;
-    options.verify = false;
-    DatabaseStatus status;
-    bilingual_str error;
-    auto database = MakeWalletDatabase(fs::PathToString(test_setup->m_path_root / "legacy"), options, status, error);
-    uint64_t create_flags = 0;
-    auto wallet = TestLoadWallet(std::move(database), context, create_flags);
+    std::unique_ptr<CWallet> wallet = std::make_unique<CWallet>(test_setup->m_node.chain.get(), "", CreateMockableWalletDatabase());
+    wallet->chainStateFlushed(ChainstateRole::NORMAL, CBlockLocator{});
+    LegacyDataSPKM* legacy_spkm = wallet->GetOrCreateLegacyDataSPKM();
 
     // Add watch-only addresses
     std::vector<CScript> scripts_watch_only;
     for (int w = 0; w < NUM_WATCH_ONLY_ADDR; ++w) {
         CKey key = GenerateRandomKey();
         LOCK(wallet->cs_wallet);
-        const CScript& script = scripts_watch_only.emplace_back(GetScriptForDestination(GetDestinationForKey(key.GetPubKey(), OutputType::LEGACY)));
-        bool res = wallet->ImportScriptPubKeys(strprintf("watch_%d", w), {script},
-                                    /*have_solving_data=*/false, /*apply_label=*/true, /*timestamp=*/1);
-        assert(res);
+        const auto& dest = GetDestinationForKey(key.GetPubKey(), OutputType::LEGACY);
+        const CScript& script = scripts_watch_only.emplace_back(GetScriptForDestination(dest));
+        assert(legacy_spkm->LoadWatchOnly(script));
+        assert(wallet->SetAddressBook(dest, strprintf("watch_%d", w), /*purpose=*/std::nullopt));
     }
 
     // Generate transactions and local addresses
-    for (int j = 0; j < 400; ++j) {
+    for (int j = 0; j < 500; ++j) {
+        CKey key = GenerateRandomKey();
+        CPubKey pubkey = key.GetPubKey();
+        Assert(legacy_spkm->LoadKey(key, pubkey));
+        // Load spendable scripts and create address book records
+        bool is_bech32 = j % 2;
+        CTxDestination dest = is_bech32 ? WitnessV0KeyHash(pubkey) : CTxDestination{PKHash(pubkey)};
+        std::string label = is_bech32 ? strprintf("bech32_%d", j) : strprintf("legacy_%d", j);
+        Assert(legacy_spkm->LoadCScript(GetScriptForDestination(dest)));
```

Comment:
This `LoadCScript` is unnecessary since the spks for public keys is either added via `ImplicitlyLearnRelatedKeyScripts` or is not in the wallet explicitly.